File: apps/apple/test-ui/WalkTests.swift
import XCTest

/// Launches the real shell and asks whether it drew its one screen – enough to prove the
/// UI test target builds, installs, and drives the app.
///
/// `@MainActor` because every XCUI type is isolated to it, while an XCTestCase method is
/// synchronous and nonisolated.
@MainActor
final class WalkTests: XCTestCase {
  private let timeout: TimeInterval = 30

  func testWalksTheShell() {
    let app = XCUIApplication()
    app.launch()

    let title = app.staticTexts["Campfire"]
    XCTAssertTrue(title.waitForExistence(timeout: timeout), "the shell's one screen")
  }
}

File: apps/android/test-ui/se/scouterna/campfire/WalkTests.kt
package se.scouterna.campfire

import androidx.lifecycle.Lifecycle
import androidx.test.ext.junit.rules.ActivityScenarioRule
import androidx.test.ext.junit.runners.AndroidJUnit4
import org.junit.Assert.assertEquals
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith

/**
 * Launches the real shell and asks whether its activity resumed – enough to prove the
 * instrumented test target builds, installs, and drives the app.
 */
@RunWith(AndroidJUnit4::class)
class WalkTests {
  @get:Rule
  val activity = ActivityScenarioRule(MainActivity::class.java)

  @Test
  fun theShellLaunches() {
    assertEquals(Lifecycle.State.RESUMED, activity.scenario.state)
  }
}
